archive/logic_puzzles/knights_and_knaves.lean

```diff
@@ -0,0 +1,82 @@
+/-
+Copyright (c) 2020 Dan Stanescu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: D. Stanescu.
```

Comment:
Perhaps use the same name formatting in both places. Otherwise someone regenerating an author list in the future is going to have to normalise by hand. :-)